docs/02-app/01-building-your-application/01-routing/07-error-handling.mdx

```diff
@@ -210,6 +213,14 @@ export default function GlobalError({ error, reset }) {
 
 ### Handling Server Errors
 
-If an error is thrown during [data fetching](/docs/app/building-your-application/data-fetching/fetching) or inside a Server Component, Next.js will forward the resulting `Error` object to the nearest `error.js` file as the `error` prop.
+If an error is thrown inside a Server Component, Next.js will forward an `Error` object (stripped of sensitive error information in production) to the nearest `error.js` file as the `error` prop.
 
-When running `next dev`, the `error` will be serialized and forwarded from the Server Component to the client `error.js`. To ensure security when running `next start` in production, a generic error message is forwarded to `error` along with a `.digest` which contains a hash of the error message. This hash can be used to correspond to server logs.
+#### Securing Sensitive Error Information
+
+During production, the `Error` instance forwarded to the client only includes a generic `message` and `digest` property.
+
+This is a security precaution to avoid leaking potentially sensitive details included in the error to the client.
+
+The `message` property contains a generic message about the error and the `digest` property contains an automatically generated hash of the error that can be used to match the corresponding error in server-side logs.
+
+During development, the `Error` instance will be serialized and forwarded from the Server Component to the `error.js` Client Component.
```

Comment:
To clarify this it'll still be a new error object, but it won't hide the `message` in order to allow you to debug.

Reply:
I've updated to be more specific.